clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -2838,6 +2838,11 @@ class SYCLKernelNameTypeVisitor
       ConstTemplateArgumentVisitor<SYCLKernelNameTypeVisitor>;
   bool IsInvalid = false;
 
+  void VisitTemplateArgs(ArrayRef<TemplateArgument> Args) {
+    for (size_t I = 0, E = Args.size(); I < E; ++I)
+      Visit(Args[I]);
```

Comment:
I wonder whether this can work:
```suggestion
    for (auto& A : Args)
      Visit(A);
```